plugins/Goals/API.php

```diff
@@ -218,6 +220,10 @@ private function checkPatternIsValid($patternType, $pattern, $matchAttribute)
         ) {
             throw new Exception(Piwik::translate('Goals_ExceptionInvalidMatchingString', array("http:// or https://", "http://www.yourwebsite.com/newsletter/subscribed.html")));
         }
+
+        if ($patternType == 'regex' && @preg_match($pattern, '') === false) {
```

Comment:
I tested it and it failed for me for `rere(.*)`. I think this is because the `/` fail. Ideally we would re-use same logic as here: https://github.com/matomo-org/matomo/blob/3.4.0/core/Tracker/GoalManager.php#L826-L836
 to ensure it will work or not work during tracking.

![image](https://user-images.githubusercontent.com/273120/39410659-01c97a0c-4c50-11e8-9726-ddba428c20f1.png)

Reply:
Good point. I've changed it :)